packages/logs/src/rawLogsEvent.types.ts

```diff
@@ -1,29 +1,67 @@
 import type { Context, ErrorSource, TimeStamp } from '@datadog/browser-core'
 import type { StatusType } from './domain/logger'
 
-export interface RawLogsEvent {
-  date?: TimeStamp
+export type RawLogsEvent =
+  | RawConsoleLogsEvent
+  | RawNetworkLogsEvent
+  | RawLoggerLogsEvent
+  | RawAgentLogsEvent
+  | RawReportLogsEvent
+  | RawRuntimeLogsEvent
+
+export type RawLogsEventOrigin = RawLogsEvent['origin']
+
+type Error = {
+  kind?: string
+  origin: ErrorSource // Todo: Remove in the next major release
+  stack?: string
+  [k: string]: unknown
+}
+
+interface CommonRawLogsEvent {
   message: string
   status: StatusType
-  origin: ErrorSource
-  logger?: {
-    name: string
-    [k: string]: unknown
-  }
-  error?: {
-    kind?: string
-    origin: ErrorSource
-    stack?: string
-    [k: string]: unknown
-  }
-  http?: {
+  error?: Error
+}
+
+interface CommonRawErrorLogsEvent extends CommonRawLogsEvent {
+  status: typeof StatusType.error
+  error: Error
+}
+
+export interface RawConsoleLogsEvent extends CommonRawLogsEvent {
+  origin: typeof ErrorSource.CONSOLE
+}
+
+export interface RawLoggerLogsEvent extends CommonRawLogsEvent {
+  origin: typeof ErrorSource.LOGGER
+}
+
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I would have love to have `logger?: {name: string}` here

Reply:
Yes me too. It directly related to https://github.com/DataDog/browser-sdk/pull/1497#discussion_r853177027
and It will be done with [RUMF-1230](https://datadoghq.atlassian.net/browse/RUMF-1230)